internal/os/iscsi/api.go

```diff
@@ -0,0 +1,192 @@
+package iscsi
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// Implements the iSCSI OS API calls. All code here should be very simple
+// pass-through to the OS APIs. Any logic around the APIs should go in
+// internal/server/iscsi/server.go so that logic can be easily unit-tested
+// without requiring specific OS environments.
+
+type APIImplementor struct{}
+
+func New() APIImplementor {
+	return APIImplementor{}
+}
+
+func (APIImplementor) AddTargetPortal(portal *TargetPortal) error {
+	cmdLine := fmt.Sprintf(
+		`New-IscsiTargetPortal -TargetPortalAddress ${Env:iscsi_tp_address} ` +
```

Comment:
The lack of `-InitiatorInstanceName` parameter assumes csi-proxy will always use the software iSCSI initiator in Windows rather any hardware NICs with iSCSI offload/support - which is fine. If you can add a comment to specify this, that would be great!

Reply:
I've added clarifying comments (in the proto), I think this is a good idea to add this in later versions.
According to Windows documentation, if you don't specify an initiator:
> "iSCSI initiator service will use any initiator that can reach the target portal."

([AddIScsiSendTargetPortalW](https://docs.microsoft.com/en-us/windows/win32/api/iscsidsc/nf-iscsidsc-addiscsisendtargetportalw))

So, at least by wording, this might suggest that the hardware initiator may be used